### PR TITLE
#3409 Make Share Modal less confusing on blueprints screen

### LINKED ISSUE #3409: Make Share Modal less confusing on blueprints screen
Context
- The current Share Modal is confusing. If the installable is already a blueprint it just shows a sharing link even if the blueprint is not shared with anyone
- The Share Modal also scales poorly with the number of teams

Implementation Sketch
- Split into 2 steps (as needed): save as blueprint, control sharing

Wireframes

Full set of wireframes here: https://balsamiq.cloud/s7ffzg5/pu105es/r2B95

image

### PR DESCRIPTION
**Pull request by BALEHOK**
## What does this PR do?

- Closes #3409

## Demo

- https://www.loom.com/share/84cf329c55454457b8b8df343ed4f242
- Convert to blueprint
Screen Shot 2022-05-20 at 9 36 59 PM

- Edit sharing
Screen Shot 2022-05-20 at 9 10 22 PM

- Read-only sharing
Screen Shot 2022-05-20 at 9 09 21 PM

## Checklist

- [ ] Add tests
- [ ] Designate a primary reviewer

### CODE REVIEW

File: src/utils/recipeUtils.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { RegistryId } from "@/core";
+import { validateRegistryId } from "@/types/helpers";
+import slugify from "slugify";
+
+/**
+ * Return a valid recipe id, or return null
```
Comment (twschiller) [inline src/utils/recipeUtils.ts:23]: This returns empty string on the catch path

Comment (BALEHOK) [reply src/utils/recipeUtils.ts:23]: This is why I'm not a big fan of comments in code.

**Comment by twschiller** [comment]
Bugs found during QA:
- [ ] In the scope dropdown in the convert modal, need to exclude options that don't have scopes defined. Currently these show up as half-height options with no label
image

File: src/utils/recipeUtils.ts (hunk 1 above)
Comment (twschiller) [reply src/utils/recipeUtils.ts:23]: 🤷  I'd prefer not to have to read the source code of every function I want to call. But yes, in general, I prefer machine-checked documentation to documentation we have to make sure is kept up-to-date manually

**Review by twschiller — APPROVED** [review]

**Comment by BALEHOK** [comment]
## Future work
Add tests
Review the usage of `getOrganizationsQuery`

@twschiller the scope dropdown is fixed at https://github.com/pixiebrix/pixiebrix-extension/pull/3474

**Pull request merged**